src/CodeGen/ODataT4CodeGenerator.tt

```diff
@@ -17,6 +17,10 @@ public static class Configuration
 
 	// The target language of the generated client code. The value must be set to "CSharp" or "VB".
 	public const string TargetLanguage = "OutputLanguage";
+	
+	// The path for the temporary file where the metadata xml document can be stored. Use this if your metadata is too big to be stored in a string literal. Ensure that you have write permission for this path.
+	// For example - "C:\\temp\\Test.xml"	
+	public const string TempFilePath = "";

```

Comment:
It sounds like this file shares the same purpose as MetadataDocumentUri with an additional purpose of letting the code gen know whether to read it as a local file. Perhaps we can have a boolean instead for that additional purpose? Please correct me if I'm misunderstanding this.

Reply:
I think they are different in the sense that MetadataDocumentUri can be a public endpoint or it can be a local file whereas the client code will not work with this as a public endpoint.

Reply:
MetadataDocumentUri can be a public endpoint or it can be a local file. Why can't we use it as the TempFilePath?

Reply:
One difference is that the header is stripped out in the temp file. The other reason is that if we tried using MetadataDocumentUri, on top of adding the stripping the header code, we would need more code to read from both file and public endpoint. Since it seemed like a low frequency issue, I did not want to change a ton for it.

Reply:
Got it, makes sense that it's different from MetadataDocumentUri then. Instead of = "", we can use = string.Empty.

Reply:
Discussed with Kanish and we'll be keeping "" instead of string.Empty due to code churn in test cases.